src/parallel.ts

```diff
@@ -25,12 +35,20 @@ export interface IBuildTrackerSettings {
  * @returns A Promise for a package build tracker.
  */
 export async function getBuildTracker(settings: IBuildTrackerSettings): Promise<ParallelBuildTracker> {
+    const normalizedSettings = {
+        fileReader: (async (filePath: string) => (await readFileWithoutBom(filePath)).toString()),
+        includeOptionalDependencies: false,
+        includePeerDependencies: false,
+        ...settings,
+    };
+
     const packagePaths = normalizePackagePaths(settings.paths);
-    const fileReader = settings.fileReader === undefined
-        ? (async (filePath: string) => (await readFileWithoutBom(filePath)).toString())
-        : settings.fileReader;
 
-    const dependencies = await getAllPackageDependencies(packagePaths, fileReader);
+    const dependencies = await getAllPackageDependencies(
+        packagePaths,
+        normalizedSettings.fileReader,
+        normalizedSettings.includeOptionalDependencies,
+        normalizedSettings.includePeerDependencies);
```

Comment:
At this point I think it'd be cleaner to change `getAllPackageDependencies` to take in a single object, so we don't have to keep track of so many parameters. Roughly:

```suggestion
    const dependencies = await getAllPackageDependencies({
        packagePaths,
        ...normalizedSettings,
    });
```

That'd also make it easier to keep having the `include*Dependencies` booleans optional, so this code wouldn't explicitly need to make them `false`.

Reply:
I updated the code to follow the pattern you suggested. However, I stuck with initializing the boolean options when I normalize. This was to prevent some funny issues tslint had with optional booleans on the `reading.ts` file.